src/components/NewCommentForm.tsx

```diff
@@ -1,6 +1,99 @@
-import React from 'react';
+import React, { FormEvent, useState } from 'react';
+import { addNewComment } from '../services/comment';
+import classNames from 'classnames';
+import { Comment } from '../types/Comment';
+
+type Props = {
+  postId: number;
+  // setComments: (comments: Comment[]) => void;
+  setComments: React.Dispatch<React.SetStateAction<Comment[]>>;
+  setError: (value: boolean) => void;
+};
+
+export const NewCommentForm: React.FC<Props> = ({
+  postId,
+  setComments,
+  setError,
+}) => {
+  const [name, setName] = useState('');
+  const [email, setEmail] = useState('');
+  const [body, setBody] = useState('');
```

Comment:
Create only one state for comment
```suggestion
  const [newComment, setNewComment] = useState({
    name: '',
    ...
  });
```